include/seqan3/search/dream_index/interleaved_bloom_filter.hpp

```diff
@@ -0,0 +1,510 @@
+     * If you want to add more bins while keeping the size constant, you need to rebuild the
+     * `seqan3::interleaved_bloom_filter`.
+     *
+     * ### Example
+     *
+     * \include test/snippet/search/dream_index/interleaved_bloom_filter_increase_bin_number_to.cpp
+     */
+    void increase_bin_number_to(bin_count const new_bins_)
+    //!\cond
+        requires data_layout_mode == data_layout::uncompressed
+    //!\endcond
+    {
+        size_t new_bins = new_bins_.get();
+
+        if (new_bins < bins)
+            throw std::invalid_argument{"The number of new bins must be >= the current number of bins."};
+
+        // Equivalent to ceil(new_bins / 64)
+        size_t new_bin_words = (new_bins + 63) >> 6;
+
+        bins = new_bins;
+        result_buffer.resize(bins);
+
+        if (new_bin_words == bin_words) // No need for internal resize if bin_words does not change.
+            return;
+
+        size_t new_technical_bins = new_bin_words << 6;
+        size_t new_bits = bin_size_ * new_technical_bins;
+
+        size_t idx_{new_bits}, idx{data.size()};
+        size_t delta = new_technical_bins - technical_bins + 64;
+
+        data.resize(new_bits);
+
+        for (size_t i = idx_, j = idx; j > 0; i -= new_technical_bins, j -= technical_bins)
+        {
+            size_t stop = i - new_technical_bins;
+
+            for (size_t ii = i - delta, jj = j - 64; stop && ii >= stop; ii -= 64, jj -= 64)
+            {
+                uint64_t old = data.get_int(jj);
+                data.set_int(jj, 0);
+                data.set_int(ii, old);
+            }
+        }
+
+        bin_words = new_bin_words;
+        technical_bins = new_technical_bins;
+    }
+    //!\}
+
+    /*!\name Lookup
+     * \{
+     */
+    /*!\brief Determines set membership of a given value.
+     * \param[in] value The raw value to process.
+     *
+     * \attention The result of this function must always be bound via reference, e.g. `auto &` to prevent copying.
+     *
+     * \details
+     *
+     * ### Example
+     *
+     * \include test/snippet/search/dream_index/interleaved_bloom_filter_bulk_contains.cpp
+     */
+    [[nodiscard]] binning_bitvector const & bulk_contains(size_t const value) const & noexcept
+    {
+        assert(result_buffer.size() == bin_count());
+
+        std::array<size_t, 5> idx;
+        std::memcpy(&idx, &hash_seeds, sizeof(size_t) * hash_funs);
+
+        for (size_t i = 0; i < hash_funs; ++i)
+        {
+            idx[i] *= value;
+            hash_and_fit(idx[i]);
+        }
+
+        for (size_t batch = 0; batch < bin_words; ++batch)
+        {
+           assert(idx[0] < data.size());
+           size_t tmp = data.get_int(idx[0]);
+           idx[0] += 64;
+
+           for (size_t i = 1; i < hash_funs; ++i)
+           {
+               assert(idx[i] < data.size());
+               tmp &= data.get_int(idx[i]);
+               idx[i] += 64;
+           }
```

Comment:
```cpp
           size_t tmp{0b1111111111111111111111111111111111111111111111111111111111111111};
           for (size_t i = 0; i < hash_funs; ++i)
           {
               assert(idx[i] < data.size());
               tmp &= data.get_int(idx[i]);
               idx[i] += 64;
           }
```

Reply:
We could also do

```
        if constexpr(data_layout_mode == data_layout::uncompressed)
        {
            std::memcpy(result_buffer.data(), data.data() + (idx[0] >> 6), technical_bins >> 3);
            for (size_t i = 1; i < hash_funs; ++i)
            {
                std::memcpy(buf.data(), data.data() + (idx[i] >> 6), technical_bins >> 3);
                result_buffer &= buf;
            }
        }
```

This gets better with a bigger bin number, breaking even at around 256 bins